adam-core/src/main/scala/org/bdgenomics/adam/rdd/feature/CoverageRDD.scala

```diff
@@ -91,8 +144,25 @@ case class CoverageRDD(rdd: RDD[Coverage],
 
   def union(rdds: CoverageRDD*): CoverageRDD = {
     val iterableRdds = rdds.toSeq
-    CoverageRDD(rdd.context.union(rdd, iterableRdds.map(_.rdd): _*),
-      iterableRdds.map(_.sequences).fold(sequences)(_ ++ _))
+
+    val mergedSequences = iterableRdds.map(_.sequences).fold(sequences)(_ ++ _)
+
+    if (iterableRdds.forall(rdd => rdd match {
+      case DatasetBoundCoverageRDD(_, _) => true
+      case _                             => false
+    })) {
+      DatasetBoundCoverageRDD(iterableRdds.map(_.dataset)
+        .fold(dataset)(_.union(_)), mergedSequences)
+    } else {
+      RDDBoundCoverageRDD(rdd.context.union(rdd, iterableRdds.map(_.rdd): _*),
```

Comment:
Yeah, however it's not clear that that is always desirable. There's two cases, and I think one of these is desirable, one isn't:

1. The _n_ RDDs have completely disjoint sequence dictionaries (e.g., you've got a single file per chromosome, and you're loading each file then unioning them). Then you can do the union a-OK.
2. The _n_ RDDs have overlapping sequence dictionaries and are co-partitioned (e.g., you have a single sorted file per sample, and want to merge across samples). Then you can do a zipPartitions and zipper-merge while preserving sort order. This maintains sort order, but has the downside of possibly creating _really large_ partitions.

Thoughts? Both are important use cases, but reducing parallelism may not be an acceptable tradeoff.